src/app/gui/queryresults/queryresultsservice.js

```diff
@@ -2097,12 +2109,12 @@ class QueryResultsService extends G3WObject {
           format,
           class: GUI.getFontClass(format),
           hint: `sdk.tooltips.download_${format}`,
-          cbk: (layer, feature, action, index) => {
+          cbk: (layer, feature, action, index, html) => {
             // un-toggle downloads action
-            this.downloadFeatures(format, layer, feature, action, index);
+            this.downloadFeatures(format, layer, feature, action, index, html);
```

Comment:
@volterra79 it doesn't work in case of single download:

## OK: [`_setActionMultiDownloadFeature`](https://github.com/g3w-suite/g3w-client/blob/bb7a36522f8bf7a40ace8216db659fdbb6b05759/src/app/gui/queryresults/queryresultsservice.js#L2101-L2121)

![image](https://github.com/g3w-suite/g3w-client/assets/9614886/f4050699-a67f-4865-8547-e594e191c31d)

![image](https://github.com/g3w-suite/g3w-client/assets/9614886/6974ea85-7260-41cb-a096-db7851c393cd)


## KO: [`saveLayerResult`](https://github.com/g3w-suite/g3w-client/blob/bb7a36522f8bf7a40ace8216db659fdbb6b05759/src/app/gui/queryresults/queryresultsservice.js#L1846-L1851)

## KO: [`_setActionDownloadFeature`](https://github.com/g3w-suite/g3w-client/blob/bb7a36522f8bf7a40ace8216db659fdbb6b05759/src/app/gui/queryresults/queryresultsservice.js#L2076-L2096)

![image](https://github.com/g3w-suite/g3w-client/assets/9614886/d29189b4-bfd0-4042-ab0f-bb9319c4ecad)

![image](https://github.com/g3w-suite/g3w-client/assets/9614886/7c43fb9b-d1c8-4d9f-ad87-807d5af20ba9)

I tried adding the `html` parameter to `this.downloadFeatures(..., html)` call, but I get cyclic object error.

Reply:
@Raruto solved https://github.com/g3w-suite/g3w-client/pull/538/commits/900ec211aafad8bc483e8c0ebb88da385c00b956

Reply:
@volterra79 there's still the same issue on [`saveLayerResult`](https://github.com/g3w-suite/g3w-client/blob/bb7a36522f8bf7a40ace8216db659fdbb6b05759/src/app/gui/queryresults/queryresultsservice.js#L1846-L1851):

![image](https://github.com/g3w-suite/g3w-client/assets/9614886/92ecf26c-2a18-44f8-baa6-b14f3c26a7ed)

Reply:
@Raruto pdf download format need to be enable only to single features and not for layers features result. I need to fix it. Thanks

Reply:
@Raruto fix with https://github.com/g3w-suite/g3w-client/pull/538/commits/471b75fe2f2a9bef984a9492f6fda777167e11b0